src/AstSemanticChecker.cpp

```diff
@@ -749,6 +756,37 @@ void AstSemanticCheckerImpl::checkClause(const AstClause& clause) {
     }
 }
 
+void AstSemanticCheckerImpl::checkMultiRule(std::set<const AstClause*> multiRule) {
+    std::map<std::string, int> var_count;
+    std::map<std::string, const AstVariable*> var_pos;
+
+    // count variable occurrences in a body only once
+    // TODO (b-scholz): for negation / disjunction this is not quite
+    // right; we would need more semantic information.
+    for (auto literal : (*multiRule.begin())->getBodyLiterals()) {
+        visitDepthFirst(*literal, [&](const AstVariable& var) {
+            var_count[var.getName()]++;
+            var_pos[var.getName()] = &var;
+        });
+    }
+    for (auto clause : multiRule) {
+        visitDepthFirst(*(clause->getHead()), [&](const AstVariable& var) {
+            var_count[var.getName()]++;
+            var_pos[var.getName()] = &var;
+        });
+    }
+
+    // check for variables only occurring once
```

Comment:
It's not clear what this is for. Why is this check needed here? The contents of the method need a little more documentation so new people can more easily understand what is being checked in `checkMultiRule`.